test/Interop/SwiftToCxx/enums/swift-enum-implementation.swift

```diff
@@ -0,0 +1,263 @@
+// CHECK-NEXT:       default: abort();
+// CHECK-NEXT:     }
+// CHECK-NEXT:   }
+// CHECK:      private:
+// CHECK:        inline char * _Nonnull _destructiveProjectEnumData() {
+// CHECK-NEXT:     auto metadata = _impl::$s5Enums1EOMa(0);
+// CHECK-NEXT:     auto *vwTableAddr = reinterpret_cast<swift::_impl::ValueWitnessTable **>(metadata._0) - 1;
+// CHECK-NEXT: #ifdef __arm64e__
+// CHECK-NEXT:     auto *vwTable = reinterpret_cast<swift::_impl::ValueWitnessTable *>(ptrauth_auth_data(reinterpret_cast<void *>(*vwTableAddr), ptrauth_key_process_independent_data, ptrauth_blend_discriminator(vwTableAddr, 11839)));
+// CHECK-NEXT: #else
+// CHECK-NEXT:     auto *vwTable = *vwTableAddr;
+// CHECK-NEXT: #endif
+// CHECK-NEXT:     const auto *enumVWTable = reinterpret_cast<swift::_impl::EnumValueWitnessTable *>(vwTable);
+// CHECK-NEXT:     enumVWTable->destructiveProjectEnumData(_getOpaquePointer(), metadata._0);
+// CHECK-NEXT:     return _getOpaquePointer();
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline void _destructiveInjectEnumTag(unsigned tag) {
+// CHECK-NEXT:     auto metadata = _impl::$s5Enums1EOMa(0);
+// CHECK-NEXT:     auto *vwTableAddr = reinterpret_cast<swift::_impl::ValueWitnessTable **>(metadata._0) - 1;
+// CHECK-NEXT: #ifdef __arm64e__
+// CHECK-NEXT:     auto *vwTable = reinterpret_cast<swift::_impl::ValueWitnessTable *>(ptrauth_auth_data(reinterpret_cast<void *>(*vwTableAddr), ptrauth_key_process_independent_data, ptrauth_blend_discriminator(vwTableAddr, 11839)));
+// CHECK-NEXT: #else
+// CHECK-NEXT:     auto *vwTable = *vwTableAddr;
+// CHECK-NEXT: #endif
+// CHECK-NEXT:     const auto *enumVWTable = reinterpret_cast<swift::_impl::EnumValueWitnessTable *>(vwTable);
+// CHECK-NEXT:     enumVWTable->destructiveInjectEnumTag(_getOpaquePointer(), tag, metadata._0);
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline unsigned _getEnumTag() const {
+// CHECK-NEXT:     auto metadata = _impl::$s5Enums1EOMa(0);
+// CHECK-NEXT:     auto *vwTableAddr = reinterpret_cast<swift::_impl::ValueWitnessTable **>(metadata._0) - 1;
+// CHECK-NEXT: #ifdef __arm64e__
+// CHECK-NEXT:     auto *vwTable = reinterpret_cast<swift::_impl::ValueWitnessTable *>(ptrauth_auth_data(reinterpret_cast<void *>(*vwTableAddr), ptrauth_key_process_independent_data, ptrauth_blend_discriminator(vwTableAddr, 11839)));
+// CHECK-NEXT: #else
+// CHECK-NEXT:     auto *vwTable = *vwTableAddr;
+// CHECK-NEXT: #endif
+// CHECK-NEXT:     const auto *enumVWTable = reinterpret_cast<swift::_impl::EnumValueWitnessTable *>(vwTable);
+// CHECK-NEXT:     return enumVWTable->getEnumTag(_getOpaquePointer(), metadata._0);
+// CHECK-NEXT:   }
+// CHECK-NEXT:   using _impl_x = decltype(x);
+// CHECK-NEXT:   using _impl_y = decltype(y);
+// CHECK-NEXT:   using _impl_z = decltype(z);
+// CHECK-NEXT:   using _impl_w = decltype(w);
+// CHECK-NEXT:   using _impl_auto = decltype(auto_);
+// CHECK-NEXT:   using _impl_foobar = decltype(foobar);
+// CHECK:      };
+// CHECK-NEXT: decltype(E::x) E::x;
```

Comment:
can you make supporting structs emitted as `inline const ` variables? that way they're actually unique to each TU and won't conflict at link time.